lib/rubocop/cop/naming/accessor_method_name.rb

```diff
@@ -63,7 +63,9 @@ def bad_reader_name?(node)
         end
 
         def bad_writer_name?(node)
-          node.method_name.to_s.start_with?('set_') && node.arguments.one?
+          node.method_name.to_s.start_with?('set_') &&
+            node.arguments.one? &&
+            !node.first_argument.forward_arg_type?
```

Comment:
This seems like it should probably only match the first `arg` type. Can you add tests and update the condition?

```
% ruby-parse -e "def foo(arg); end"
(def :foo
  (args
    (arg :arg)) nil)
```

```
% ruby-parse -e "def foo(arg = :default); end"
(def :foo
  (args
    (optarg :arg
      (sym :default))) nil)
```

```
% ruby-parse -e "def foo(*args); end"
(def :foo
  (args
    (restarg :args)) nil)
```

```
% ruby-parse -e "def foo(k:); end"
(def :foo
  (args
    (kwarg :k)) nil)
```

```
% ruby-parse -e "def foo(**options); end"
(def :foo
  (args
    (kwrestarg :options)) nil)
```

```
% ruby-parse -e "def foo(&block); end"
(def :foo
  (args
    (blockarg :block)) nil)
```

```
% ruby-parse -e "def foo(...); end"
(def :foo
  (args
    (forward-arg)) nil)
```

Reply:
Yes, that sounds very good to me. I'll fix it.